.devcontainer/devcontainer.json

```diff
@@ -38,29 +38,62 @@
 
     "containerEnv": {
         "HOST_MORPHEUS_ROOT": "${localWorkspaceFolder}",
-        "MORPHEUS_ROOT": "${containerWorkspaceFolder}"
+        "MORPHEUS_ROOT": "${containerWorkspaceFolder}",
+        "DEFAULT_CONDA_ENV": "morpheus",
+        "MAMBA_NO_BANNER": "1",
+        "VAULT_HOST": "https://vault.ops.k8s.rapids.ai",
+        "DISPLAY": "${localEnv:DISPLAY:-}",
+        "XAUTHORITY": "${localEnv:XAUTHORITY:-}",
+        "XDG_SESSION_TYPE": "${localEnv:XDG_SESSION_TYPE:-}",
+        "XDG_RUNTIME_DIR": "${localEnv:XDG_RUNTIME_DIR:-}",
+        "DBUS_SESSION_BUS_ADDRESS": "${localEnv:DBUS_SESSION_BUS_ADDRESS:-}"
     },
 
-    "postStartCommand": "${containerWorkspaceFolder}/.devcontainer/bin/setup-morpheus-env",
-
-    "initializeCommand": [ "./.devcontainer/init.sh" ],
+    "initializeCommand": [ "${localWorkspaceFolder}/.devcontainer/initialize-command.sh" ],
 
     "remoteUser": "coder",
-
     "mounts": [
-        {"type":"bind", "source": "/var/run/docker.sock", "target": "/var/run/docker.sock"},
-        {"type":"bind", "source": "${localWorkspaceFolder}/.cache/conda", "target": "/home/coder/.conda"},
-        {"type":"bind", "source": "${localEnv:HOME}/.config/gh", "target": "/home/coder/.config/gh"}
+        {
+            "type": "bind",
+            "source": "/tmp/.X11-unix",
+            "target": "/tmp/.X11-unix"
+        },
+        {
+            "type": "bind",
+            "source": "${localEnv:XDG_RUNTIME_DIR}",
+            "target": "${localEnv:XDG_RUNTIME_DIR}"
+        },
+        {
+            "type": "bind",
+            "source": "/run/dbus/system_bus_socket",
+            "target": "/run/dbus/system_bus_socket"
+        },
+        {
+            "type":"bind",
+            "source": "/var/run/docker.sock",
+            "target": "/var/run/docker.sock"
+        },
+        {
+            "type":"bind",
+            "source": "${localWorkspaceFolder}/../.conda",
+            "target": "/home/coder/.conda"
+        },
```

Comment:
Is there a reason you need to use different environment names? I ask because using different env names defeats sccache.  For example, I'm using `legate` for legate + cunumeric, and `rapids` for the various RAPIDS library devcontainers.

I mount separate host dirs to `~/.conda/envs` for the RAPIDS libs, so that devcontainers for separate repos don't overwrite each other, but they all use the env name "rapids". That way building `libcudf` from source in the cuDF devcontainer (using RMM from conda) will cache the same as building it from source in the cuSpatial devcontainer.

Reply:
Another reason I use the same env name is that deleting + recreating (or updating via `mamba env update`) is fairly quick assuming the packages are cached.

Reply:
Any reason why we wouldn't want separate mounting points for pkgs and envs? Seems fine to me, and less confusing than keying the env name off the directory.

Reply:
mounting `.conda/envs` and `.conda/pkgs` separately is going to cause a permissions error on `.conda`, which will prevent conda from creating the test and envinroments.yml file.

Reply:
Can we do something like this?
```
        {
            "type": "bind",
            "source": "${localWorkspaceFolder}/.conda",
            "target": "/home/coder/.conda"
        },
        {
            "type": "bind",
            "source": "${localEnv:CONDA_PREFIX:-${localWorkspaceFolder}/.conda}/pkgs",
            "target": "/home/coder/.conda/pkgs"
        },
```

Reply:
it appears shell expansions don't work here, so `:-<default>` isn't going to help. And we can't assume the host has conda installed, so `CONDA_PREFIX` isn't going to work this way.

We might be able to create a `.env` file in the devcontainer directory as part of the initialize command, though? Anyone feel any which way about that?

Reply:
I went ahead and used two separate mounts.

Reply:
That combination isn't valid because you can't bind mount a host dir into an existing bind mount. I think I've done that before with overlays or something, but I can't recall.

Either way, that's not what I'm recommending. This is:


```
# host paths:
project-1/...
project-2/...
.conda/pkgs
.conda/envs-project-1/morpheus
.conda/envs-project-2/morpheus

# project-1/.devcontainer.json:
initializeCommand: ["/bin/bash", "-c", "mkdir -m 0755 -p ${localWorkspaceFolder}/../.conda/{pkgs,${localWorkspaceFolderBasename}-envs}"],
mounts: [
  "source=${localWorkspaceFolder}/../.conda/pkgs,target=/home/coder/.conda/pkgs,type=bind"
  "source=${localWorkspaceFolder}/../.conda/envs-${localWorkspaceFolderBasename},target=/home/coder/.conda/envs,type=bind"
]

# project-2/.devcontainer.json:
initializeCommand: ["/bin/bash", "-c", "mkdir -m 0755 -p ${localWorkspaceFolder}/../.conda/{pkgs,${localWorkspaceFolderBasename}-envs}"],
mounts: [
  "source=${localWorkspaceFolder}/../.conda/pkgs,target=/home/coder/.conda/pkgs,type=bind"
  "source=${localWorkspaceFolder}/../.conda/envs-${localWorkspaceFolderBasename},target=/home/coder/.conda/envs,type=bind"
]
```

Using `${localWorkspaceFolderBasename}` in the bind mounts in both projects' devcontainer JSON files will expand out to `project-1`, so `../.conda/envs-project-1` is mounted into `/home/coder/.conda/envs` for the project-1 devcontainer, and vice-versa for the `project-2` devcontainer.

So doing that, they can both can  share the package cache, have separate conda envs named `morpheus`, while still keeping a consistent path in compile commands for build caches (i.e. `-I/home/coder/.conda/envs/morpheus/include`).